src/components/common/ColonyHome/ExtensionUpgrade/ExtensionUpgrade.tsx

```diff
@@ -26,25 +29,19 @@ const MSG = defineMessages({
 
 const ExtensionUpgrade = () => {
   const { colony } = useColonyContext();
-  const { colonyAddress, name } = colony || {};
+  const { name } = colony || {};
 
   const { extensionId } = useParams<{
     extensionId: string;
   }>();
 
-  const { data } = useColonyExtensionsQuery({
-    variables: { address: colonyAddress },
-  });
+  const { extensionData } = useExtensionData(Extension.OneTxPayment);
 
   const isExtensionDetailsRoute = extensionId === Extension.OneTxPayment;
 
-  const oneTxPaymentExtension = data?.processedColony?.installedExtensions.find(
-    ({ details, extensionId: extensionName }) =>
-      details?.initialized &&
-      !details?.missingPermissions.length &&
-      extensionName === Extension.OneTxPayment,
+  const mustUpgrade = useOneTxMustBeUpgraded(
+    extensionData as InstalledExtensionData,
```

Comment:
Just checking this is how it works in the Dapp? That the banner only appears if the extension has been installed?

Is it not also possible to upgrade an extension that hasn't been installed yet? I.e. so that you install the most up to date version?

Reply:
Should also include the `| null` typing from `extensionData`

Reply:
@Julianahlk I think this is still outstanding. The check is one-line, so we shouldn't need a separate function making extra query calls